### ISSUE TITLE
comEle = _AutomationClient.instance().dll.GetNextSiblingElement(self.Element) Hangs

### ISSUE THREAD

**Issue by akshaykochar** [yinkaisheng/Python-UIAutomation-for-Windows#30]
I was running find control. The function hung in this spot. This is the code I used.
>>> import uiautomation
>>> for item, depth in uiautomation.WalkControl(uiautomation.GetRootControl(), maxDepth=1):
	     print item

**Comment by akshaykochar** [comment]
Also tried this code. It get stuck.
>>> import uiautomation
>>> r = uiautomation.GetRootControl()
>>> r.GetChildren()

**Comment by yinkaisheng** [comment]
I had met this problem too on a PC a few months ago.
I found that Microsoft's Inspect.exe also had this problem too on that PC.
It happens when call GetNextSiblingControl(GetNextSiblingElement) on root's last child.
It should return None but it hangs.
It may disappear after a reboot. May be a bug of UIAutomation.

**Comment by akshaykochar** [comment]
Yes you are right. A reboot or restart of python fixes the issue. Any way of updating the dll code?

**Comment by yinkaisheng** [comment]
I'm afraid not.

**Comment by akshaykochar** [comment]
@yinkaisheng, this is happening again on one of my PC's, do you know the best way to fix this issue. How did you fix it. I have restarted the machine a bunch of times but it still occurs.

**Comment by yinkaisheng** [comment]
Sorry, I have no idea. 
Microsoft's Inspect.exe hangs too. It hangs in the Microsoft's API, not my code.
I don't know how to fix it.

**Comment by tutufan519** [comment]
Oh my mother, I have the same problem~

**Comment by shairpapy** [comment]
Today is February 10, 2026, and this problem still exists.

**Comment by TeamDman** [comment]
I was experiencing something like this while using UI automation via Rust.
I think it's a problem with the Windows implementation; there are situations where you just have to create your own heuristics.

For example, when iterating children, 

> It happens when call GetNextSiblingControl(GetNextSiblingElement) on root's last child.
It should return None but it hangs.

You can mitigate the problem by getting the ID of the last element and intentionally stopping iteration once reached (never calling `next` on the last element) or by hardcoding the class/name of the last element.

```rust
    fn should_stop(&self, element: &UIElement) -> bool {
        element.get_name() == Ok("Program Manager".to_string())
            && element.get_classname() == Ok("Progman".to_string())
        // This could be more specific, but until a false positive is encountered, this is fine
    }
```

This worked pretty well for me, just gotta treat the underlying stuff as a bit hostile and be ready to hardcode your own mitigations

https://github.com/TeamDman/Cursor-Hero/blob/9ba3be8c66d468f0d3e54da5e6b1b4cd0e7718e5/crates/ui_automation/src/gather_children.rs#L52-L58
